Research/Research/RSDTaskMetadata.swift

```diff
@@ -0,0 +1,239 @@
+        }
+        return identifier
+    }
+    
+    /// A human-readable mapped name for a given device type.
+    fileprivate var machineName: String {
+        let identifier = deviceTypeIdentifier
+        switch identifier {
+        case "Watch1":                                      return "Apple Watch Series 1"
+        case "Watch2,6","Watch2,7","Watch2,3","Watch2,4":   return "Apple Watch Series 2"
+        case "Watch3,1","Watch3,2","Watch3,3","Watch3,4":   return "Apple Watch Series 3"
+        case "i386", "x86_64":                              return "Apple Watch Simulator"
+            
+        default:                                            return identifier
+        }
+    }
+}
+
+#elseif os(macOS)
+
+import AppKit
+
+#else
+import UIKit
+extension UIDevice {
+    
+    /// An identifier for the device type pulled from the system info.
+    fileprivate var deviceTypeIdentifier: String {
+        var systemInfo = utsname()
+        uname(&systemInfo)
+        let machineMirror = Mirror(reflecting: systemInfo.machine)
+        let identifier = machineMirror.children.reduce("") { identifier, element in
+            guard let value = element.value as? Int8, value != 0 else { return identifier }
+            return identifier + String(UnicodeScalar(UInt8(value)))
+        }
+        return identifier
+    }
+    
+    /// A human-readable mapped name for a given device type.
+    fileprivate var machineName: String {
+        let identifier = deviceTypeIdentifier
+        switch identifier {
+        case "iPod5,1":                                     return "iPod Touch 5"
+        case "iPod7,1":                                     return "iPod Touch 6"
+            
+        case "iPhone3,1", "iPhone3,2", "iPhone3,3":         return "iPhone 4"
+        case "iPhone4,1":                                   return "iPhone 4s"
+        case "iPhone5,1", "iPhone5,2":                      return "iPhone 5"
+        case "iPhone5,3", "iPhone5,4":                      return "iPhone 5c"
+        case "iPhone6,1", "iPhone6,2":                      return "iPhone 5s"
+        case "iPhone7,2":                                   return "iPhone 6"
+        case "iPhone7,1":                                   return "iPhone 6 Plus"
+        case "iPhone8,1":                                   return "iPhone 6s"
+        case "iPhone8,2":                                   return "iPhone 6s Plus"
+        case "iPhone9,1", "iPhone9,3":                      return "iPhone 7"
+        case "iPhone9,2", "iPhone9,4":                      return "iPhone 7 Plus"
+        case "iPhone8,4":                                   return "iPhone SE"
+        case "iPhone10,1", "iPhone10,4":                    return "iPhone 8"
+        case "iPhone10,2", "iPhone10,5":                    return "iPhone 8 Plus"
+        case "iPhone10,3", "iPhone10,6":                    return "iPhone X"
+            
```

Comment:
I wish Apple provided this mapping -- at some point we should find out and add the mappings for iPhone XS, XS Max, and XR (and I'm guessing there are some new iPads too since we last updated this)